src/psyclone/domain/lfric/kernel/field_vector_arg.py

```diff
@@ -89,14 +91,21 @@ def create_from_fparser2(fparser2_tree):
         :returns: an instance of this class.
         :rtype: :py:class:`psyclone.domain.lfric.kernel.FieldVectorArg`
 
-        :raises TypeError: if the first metadata argument is not in \
-            the form datatype*vector_length.
+        raises ValueError: if the metadata is not in the correct form.
 
         '''
         FieldVectorArg.check_fparser2(fparser2_tree, nargs=4)
-        vector_length = FieldVectorArg.get_vector_length(fparser2_tree)
+        FieldVectorArg.check_first_arg(
```

Comment:
Again, I think you could shorten this to `self` if you want.

Reply:
This is within the same static method so there is no 'self'.